source/beamline/BioXAS/BioXASBeamline.cpp

```diff
@@ -117,12 +127,13 @@ AMAction3* BioXASBeamline::createScanInitializationAction(AMGenericStepScanConfi
 		if (zebraInitialization)
 			initializationAction->addSubAction(zebraInitialization);
 
-		// Initialize the scaler.
+		// Initialize the scaler, if using.
 
 		AMListAction3 *scalerInitialization = 0;
 		CLSSIS3820Scaler *scaler = BioXASBeamline::bioXAS()->scaler();
 
-		if (bioXASGenericStepScanConfiguration && bioXASGenericStepScanConfiguration->usingScaler()) {
```

Comment:
is the scaler used for All the generic scans? Based on your original implementation of the `usingScaler()` function, we are using Scaler only if we include `i0 || i1 || i2`